src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -645,6 +649,11 @@ protected override TypeSymbol InferTypeOfVarVariable(DiagnosticBag diagnostics)
                         loopBinder.BindDeconstructionDeclaration(forStatement.Declaration, forStatement.Declaration, diagnostics);
                         break;
 
+                    case SyntaxKind.ForEachStatement:
+                        var foreachBinder = this.binder.GetBinder((ForEachStatementSyntax)statement);
+                        foreachBinder.BindForEachDeconstruction(diagnostics, foreachBinder);
```

Comment:
Isn't `this.binder` the `ForEachBinder`? If so, `ForEachBinder.BindForEachDeconstruction` can be called directly and the method does not need to be `virtual`.